<a href="https://colab.research.google.com/github/toupasana/Exploratory-Data-Analysis/blob/main/Copy_of_Play_Store_App_Review_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

###Index Of Contents
####· Introduction
####· Data Preparation and Cleaning
####· Exploratory Analysis and Visualization
  ∘ Let’s have a look at the distribution of the ratings of the data frame.

  ∘ Let’s plot a visualization graph to view what portion of the apps in the play store are paid and free.

  ∘ Which category App’s have the most number of installs?

####· Asking and Answering Questions
  ∘ What are the Top 10 installed apps in any category?

  ∘ Which are the top 10 expensive Apps in the play store?

  ∘ Which are the Apps with the highest number of reviews?

  ∘ What are the count of Apps in different genres?
  
  ∘ Which are the apps that have made the highest-earning?
####· Inferences and Conclusion
####· References

* Let us start by importing some of the required libraries with which we will be working on.



In [3]:
 
import pandas as pd
import numpy as np

In [ ]:
import warnings
warnings.filterwarnings('ignore')

* Loading the dataset as a pandas data frame.



In [4]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [5]:
playstore_df =  pd.read_csv('/content/drive/MyDrive/EDA/Playstore/Play Store Data.csv')

* After loading the dataset, we can start the exploration but before that, we need to check and see that the dataset is ready for performing several exploration operations or not, so let’s first have a look at the structure and the manner in which the data is organized.

In [6]:
playstore_df.head(10)

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   
5                         Paper flowers instructions  ART_AND_DESIGN     4.4   
6            Smoke Effect Photo Maker - Smoke Editor  ART_AND_DESIGN     3.8   
7                                   Infinite Painter  ART_AND_DESIGN     4.1   
8                               Garden Coloring Book  ART_AND_DESIGN     4.4   
9                      Kids Paint Free - Drawing Fun  ART_AND_DESIGN     4.7   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   
5     167  5.6M      50,000+  Free     0       Everyone   
6     178   19M      50,000+  Free     0       Everyone   
7   36815   29M   1,000,000+  Free     0       Everyone   
8   13791   33M   1,000,000+  Free     0       Everyone   
9     121  3.1M      10,000+  Free     0       Everyone   

                      Genres        Last Updated         Current Ver  \
0               Art & Design     January 7, 2018               1.0.0   
1  Art & Design;Pretend Play    January 15, 2018               2.0.0   
2               Art & Design      August 1, 2018               1.2.4   
3               Art & Design        June 8, 2018  Varies with device   
4    Art & Design;Creativity       June 20, 2018                 1.1   
5               Art & Design      March 26, 2017                 1.0   
6               Art & Design      April 26, 2018                 1.1   
7               Art & Design       June 14, 2018            6.1.61.1   
8               Art & Design  September 20, 2017               2.9.2   
9    Art & Design;Creativity        July 3, 2018                 2.8   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up  
5    2.3 and up  
6  4.0.3 and up  
7    4.2 and up  
8    3.0 and up  
9  4.0.3 and up

In [12]:
print(playstore_df.shape)
print(playstore_df.columns)

(10841, 13)
Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')


The dataset is been consists of 13 columns:- App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver, and Android Ver with 10841 Rows.

In [13]:

#To know if there is any missing value or Nan value in the dataset, we can use the isnull() function.
playstore_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

We can see above that the dataset contains many Null or missing values. The column Rating, Type , Content Rating , Current Ver , andAndroid Ver contains 1474, 1, 1, 8, and 3 missing values respectively.

So, we will need to prepare the dataset before performing exploratory data analysis on it.

###Data Preparation and Cleaning
Data preparation is the process of cleaning and transforming raw data prior to processing and analysis. It is an important step prior to processing and often involves reformatting data, making corrections to data, and the combining of data sets to enrich data. Data cleansing or data cleaning is the process of detecting and correcting (or removing) corrupt or inaccurate records from a recordset, table, or database and refers to identifying incomplete, incorrect, inaccurate, or irrelevant parts of the data and then replacing, modifying, or deleting the dirty or coarse data.

* Here we are defining a function to get more useful information about the different attributes of the dataset, also there is one more valid point in defining a function which it will be reusable, and we are going to utilize our defined function several times in future.



In [14]:
def printinfo():
    temp = pd.DataFrame(index=playstore_df.columns)
    temp['data_type'] = playstore_df.dtypes
    temp['null_count'] = playstore_df.isnull().sum()
    temp['unique_count'] = playstore_df.nunique()
    return temp

In [15]:
#Let’s call the function and see what it returns:

printinfo()

data_type  null_count  unique_count
App               object           0          9660
Category          object           0            34
Rating           float64        1474            40
Reviews           object           0          6002
Size              object           0           462
Installs          object           0            22
Type              object           1             3
Price             object           0            93
Content Rating    object           1             6
Genres            object           0           120
Last Updated      object           0          1378
Current Ver       object           8          2832
Android Ver       object           3            33

* we can now see the missing number of values of any attribute, its unique count, and its respective data types.

 * So we can start the process of data cleaning, lets start with the column Type

In [16]:
playstore_df[playstore_df.Type.isnull()]

App Category  Rating Reviews                Size  \
9148  Command & Conquer: Rivals   FAMILY     NaN       0  Varies with device   

     Installs Type Price Content Rating    Genres   Last Updated  \
9148        0  NaN     0   Everyone 10+  Strategy  June 28, 2018   

             Current Ver         Android Ver  
9148  Varies with device  Varies with device

There is only one missing value in this column, So, let’s fill the missing value. After cross-checking in the play store the missing value is found to be Free, So now we can fill the missing value with free .

In [17]:
playstore_df['Type'].fillna("Free", inplace = True)

After filling the value we can check and see if that has been correctly placed.

In [18]:
playstore_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 0
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

Now, we can move on to the column Content Rating :

In [19]:
playstore_df[playstore_df['Content Rating'].isnull()]

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

By, looking only at these rows it is not easy to say what's actually missing in this row. let us have a look at all of its near rows data. Using ,iloc and loc function.



In [20]:
playstore_df.loc[10468:10477, :]

App         Category  Rating  \
10468                         Tassa.fi Finland        LIFESTYLE     3.6   
10469              TownWiFi | Wi-Fi Everywhere    COMMUNICATION     3.9   
10470                               Jazz Wi-Fi    COMMUNICATION     3.4   
10471                         Xposed Wi-Fi-Pwd  PERSONALIZATION     3.5   
10472  Life Made WI-Fi Touchscreen Photo Frame              1.9    19.0   
10473                  osmino Wi-Fi: free WiFi            TOOLS     4.2   
10474                             Sat-Fi Voice    COMMUNICATION     3.4   
10475                         Wi-Fi Visualizer            TOOLS     3.9   
10476                    Lennox iComfort Wi-Fi        LIFESTYLE     3.0   
10477              Sci-Fi Sounds and Ringtones  PERSONALIZATION     3.6   

      Reviews    Size     Installs  Type     Price Content Rating  \
10468     346    7.5M      50,000+  Free         0       Everyone   
10469    2372     58M     500,000+  Free         0       Everyone   
10470      49    4.0M      10,000+  Free         0       Everyone   
10471    1042    404k     100,000+  Free         0       Everyone   
10472    3.0M  1,000+         Free     0  Everyone            NaN   
10473  134203    4.1M  10,000,000+  Free         0       Everyone   
10474      37     14M       1,000+  Free         0       Everyone   
10475     132    2.6M      50,000+  Free         0       Everyone   
10476     552    7.6M      50,000+  Free         0       Everyone   
10477     128     11M      10,000+  Free         0       Everyone   

                  Genres        Last Updated Current Ver   Android Ver  
10468          Lifestyle        May 22, 2018         5.5    4.0 and up  
10469      Communication      August 2, 2018       4.2.1    4.2 and up  
10470      Communication   February 10, 2017         0.1    2.3 and up  
10471    Personalization      August 5, 2014       3.0.0  4.0.3 and up  
10472  February 11, 2018              1.0.19  4.0 and up           NaN  
10473              Tools      August 7, 2018     6.06.14    4.4 and up  
10474      Communication   November 21, 2014     2.2.1.5    2.2 and up  
10475              Tools        May 17, 2017       0.0.9    2.3 and up  
10476          Lifestyle      March 22, 2017      2.0.15  2.3.3 and up  
10477    Personalization  September 27, 2017         4.0    4.0 and up

* We can clearly see that row 10472 has missing data for the Categorycolumn and all the prevailing column values are being replaced with its previous column. A better idea will be to drop this row from our data frame.

In [21]:
playstore_df.dropna(subset = ['Content Rating'], inplace=True)

We are having some of the unwanted columns which will be of not much use in the analysis process. So let’s drop those columns.

In [22]:
playstore_df.drop(['Current Ver','Last Updated', 'Android Ver'], axis=1, inplace=True)

* Now, we can fix the Rating column which contains a total of 1474 of missing values. Replacing the missing values with the Modevalue of that entire column.

In [23]:
modeValueRating = playstore_df['Rating'].mode()

In [24]:
playstore_df['Rating'].fillna(value=modeValueRating[0], inplace = True)

* Finally, after fixing all the missing values, we should have a look at our data frame, We defined a function as printinfo() . So, it’s time to use that function.

In [25]:

printinfo()

data_type  null_count  unique_count
App               object           0          9659
Category          object           0            33
Rating           float64           0            39
Reviews           object           0          6001
Size              object           0           461
Installs          object           0            21
Type              object           0             2
Price             object           0            92
Content Rating    object           0             6
Genres            object           0           119

All the columns have the null_count as zero, which indicates that now the data frame doesn’t contain any missing values.

* Now we are done with the data cleansing part and in a state to start the work for <u>data preparation</u>

* Columns like Reviews, Size, Installs, & priceshould have an intor floatdatatype, But here we can see of objecttype, So let’s convert them to their respective correct type.

In [ ]:
#Starting with the column Reviews , converting its type to int .
playstore_df['Reviews'] = playstore_df.Reviews.astype(int)